### PR TITLE
Add Pancake Sort in Kotlin

### LINKED ISSUE #2151: Pancake Sort
- [x] C
- [x] C++
- [x] C# - @himanshusharma89
- [ ] CoffeeScript
- [ ] Dart - @Rukmini-Meda
- [x] Go - @KamalDGRT
- [ ] Java - @parjanyaacoder
- [x] JS - @sejalshri
- [ ] Kotlin - @sandhyabhan
- [ ] Php - @sakshikhachane
- [x] Python
- [ ] Readme - @Tawishi
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by sandhyabhan**
Fixes #2151 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Add Pancake Sort in Kotlin

#### Languages Used:
- Kotlin

#### Files Added:
- Pancake_Sort.kt

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Pancake_Sort/Pancake_Sort.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+       	    array[i] = read.nextLine().toInt()
+        }
+        pancakesort ( array , n ) ;
+        for (i in 0 until n)
+	{
+            println( array[i] + " " ) ;	
+	}
+        
+    }
+
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:69]: Remove this line

File: Pancake_Sort/Pancake_Sort.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        for(i in 0 until n)
+   	{
+       	    array[i] = read.nextLine().toInt()
+        }
+        pancakesort ( array , n ) ;
+        for (i in 0 until n)
+	{
+            println( array[i] + " " ) ;	
+	}
+
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:67]: remove this line

File: Pancake_Sort/Pancake_Sort.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        int i , j , k , l , m , n ; 
+        val n = read.nextLine().toInt()
+        int array [] = new int [n] ;
+        for(i in 0 until n)
+   	{
+       	    array[i] = read.nextLine().toInt()
+        }
+        pancakesort ( array , n ) ;
+        for (i in 0 until n)
+	{
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:64]: Inconsistency in indentation

File: Pancake_Sort/Pancake_Sort.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    
+    
+    fun main ()
+    {
+        var read = Scanner(System.`in`)
+        int i , j , k , l , m , n ; 
+        val n = read.nextLine().toInt()
+        int array [] = new int [n] ;
+        for(i in 0 until n)
+   	{
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:59]: Inconsistency in indentation

File: Pancake_Sort/Pancake_Sort.kt [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        for (currentLength in n until 0)
+        {
+            int j = findmax (a , currentLength-1) ;
+            if (j != currentLength)
+            {
+                reverseArray (a , j) ;
+                reverseArray (a , currentLength-1) ;
+            }
+        }
+     }
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:49]: Inconsistency in indentation

File: Pancake_Sort/Pancake_Sort.kt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    // This functions reverses the array upto index n
+    fun reverseArray ( int:a [] , int:n ):void
+    {
+        int temp[] = new int [n+1] ;   
+        for (i in 0 until n)
+        {
+            temp[i] = a[n-i];
+        }
+        for (i in 0 until n)
+	{
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:14]: Inconsistency in indentation

File: Pancake_Sort/Pancake_Sort.kt [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+         {  
+             if (a[i] > max)
+             {
+                 max = a [i] ;
+                 j = i ;
+             }
+         }
+              
+         return j ; 
+
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:33]: Remove this line

File: Pancake_Sort/Pancake_Sort.kt [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+             {
+                 max = a [i] ;
+                 j = i ;
+             }
+         }
+              
+         return j ; 
+                
+     }
+
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:35]: remove this line

File: Pancake_Sort/Pancake_Sort.kt [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            int j = findmax (a , currentLength-1) ;
+            if (j != currentLength)
+            {
+                reverseArray (a , j) ;
+                reverseArray (a , currentLength-1) ;
+            }
+        }
+    }
+
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:50]: remove this line

File: Pancake_Sort/Pancake_Sort.kt (hunk 4 above)
Comment (somya-kapoor) [reply Pancake_Sort/Pancake_Sort.kt:59]: Inconsistency in indentation

File: Pancake_Sort/Pancake_Sort.kt (hunk 3 above)
Comment (somya-kapoor) [reply Pancake_Sort/Pancake_Sort.kt:64]: Inconsistency in indentation

File: Pancake_Sort/Pancake_Sort.kt [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        int array [] = new int [n] ;
+        for(i in 0 until n)
+   	{
+       	    array[i] = read.nextLine().toInt()
+        }
+        pancakesort (array, n) ;
+        for (i in 0 until n)
+	{
+            println( array[i] + " " ) ;	
+	}
```
Comment (somya-kapoor) [inline Pancake_Sort/Pancake_Sort.kt:66]: Inconsistency in indentation

**Comment by sandhyabhan** [comment]
I cannot see some of the changes that you have marked.

**Comment by sandhyabhan** [comment]
I have made the asked changes. Pls check.

File: Pancake_Sort/Pancake_Sort.kt [hunk 11]
```diff
@@ -0,0 +1,71 @@
+// Kotlin Code for Pancake Sort
+
+class PancakeSorting
+{   
+    // This functions reverses the array upto index n
+    fun reverseArray ( int:a [] , int:n ):void
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.kt:6]: please remove the space before comma

File: Pancake_Sort/Pancake_Sort.kt [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        int temp[] = new int [n+1] ;   
+        for (i in 0 until n)
+        {
+            temp[i] = a[n-i];
+        }
+        for (i in 0 until n)
+        {
+        	a[i] = temp[i];
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.kt:15]: correct the indentation

**Comment by sandhyabhan** [comment]
I cannot see the above errors.
I tried using another code editor too.

**Comment by asha15** [comment]
try to edit in GitHub

**Pull request merged**